### PR TITLE
BOAC-2692: Adds SIDs to manually_added_advisees when searched or curated

### PR DESCRIPTION
**Pull request by lyttam**
https://jira.ets.berkeley.edu/jira/browse/BOAC-2692

### CODE REVIEW

File: boac/merged/student.py [hunk 1]
```diff
@@ -464,6 +467,7 @@ def search_for_students(
 
 
 def search_for_student_historical(sid):
+    ManuallyAddedAdvisee.find_or_create(sid)
```
Comment (paulkerschen) [inline boac/merged/student.py]: @lyttam This should probably go inside the `if profile` condition below? At this point in the code we don't yet know if it's a real SID or any old string of numbers.

**Comment by paulkerschen** [comment]
Code so simple indicates sensible structures around it! I think one adjustment is advisable before merge.

File: boac/merged/student.py (hunk 1 above)
Comment (lyttam) [reply boac/merged/student.py]: @paulkerschen I wasn't sure if the profile could be missing for a valid reason - but we definitely don't want any old strings of numbers being written to a table. I'll move it.

**Comment by lyttam** [comment]
So simple it threw me for a loop initially!

**Pull request merged**